version: '3'

services:

  db:
    image: mysql:5.7
    command:  --innodb-use-native-aio=0
    container_name: db
    restart: always
    tty: true
    volumes:
      - ./init-scripts:/docker-entrypoint-initdb.d
      - ./mysql:/var/lib/mysql
    environment:
      - MYSQL_DATABASE=nodedb
      - MYSQL_ROOT_PASSWORD=root
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "-p$$MYSQL_ROOT_PASSWORD"]
      interval: 5s
      timeout: 10s
      retries: 3
    networks:
      - challenge2

  app:
    build: 
      context: node
      dockerfile: Dockerfile
    container_name: app
    networks:
      - challenge2
    volumes:
      - ./node:/usr/src/app
    tty: true
    ports:
      - "3000:3000"
    depends_on:
      db:
        condition: service_healthy

  nginx:
    build: 
      context: nginx
      dockerfile: Dockerfile
    container_name: nginx
    tty: true
    ports:
      - "8080:80"
    networks:
      - challenge2
    links:
      - app
    depends_on:
      - app

networks:
  challenge2:
    driver: bridge